PinClustering/PinClustering/Model/Managers/NetworkManager.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  NetworkManager.swift
+//  PinClustering
+//
+//  Created by Norbert Agoston on 06/09/16.
+//  Copyright © 2016 Norbert Agoston. All rights reserved.
+//
+
+import UIKit
+
+class NetworkManager: NSObject {
+    
+    static let sharedInstance = NetworkManager()
+    
+    func httpGet(url: NSURL, callback: (NSArray?, String?) -> Void) {
+        let configuration =  NSURLSessionConfiguration.defaultSessionConfiguration()
+        let session = NSURLSession(configuration: configuration)
+        let request = NSMutableURLRequest(URL: url)
+        
+        let task = session.dataTaskWithRequest(request) { (data, response, error) in
+            do {
+                if error != nil {
+                    callback(nil,error?.localizedDescription)
+                } else {
+                    if let responseData = data {
+                        let json = try NSJSONSerialization.JSONObjectWithData(responseData, options: .AllowFragments)
+                        
+                        if let locations = json as? [[String: AnyObject]] {
+                            callback(locations,nil)
```

Comment:
the callback could never be called if the json is not correctly formatted. The proper way to handle this is to give the receiver  the options to handle the data as they want.